### PR TITLE
nixos/smartd: add option for notifiction email sender

### PR DESCRIPTION
**Pull request by tmplt**
###### Motivation for this change

I have a dedicated robots@domain.tld email I want to use when automatically reporting errors to myself. The current configuration of `services.smartd` requires one to send an email from root@domain.tld if `services.smartd.notifictions.mail.sender` is configured for msmtp.

This new option allows the sender in the email's `From` field to be configured, and uses `root` as default, mirroring present behavior.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by tmplt** [comment]
This is a minor change. Should it be merged into `release-20.03` instead?

File: nixos/modules/services/monitoring/smartd.nix [hunk 1]
```diff
@@ -129,6 +129,12 @@ in
             description = "Whenever to send e-mail notifications.";
           };
 
+          sender = mkOption {
+            default = "root";
```
Comment (tmplt) [inline nixos/modules/services/monitoring/smartd.nix:133]: Add an `example = robots@domain.tld`.

**Comment by dasJ** [comment]
Backporting doesn't really work that way. The BACK part of backporting menas it's already on master. If you just merged this into 20.03, it wouldn't be on master, and would hence be missing on 20.09. Therefore, you need to target master with this PR and backport it in a second PR after this one is merged.

Also, this is not really a backport-worthy change imo, since it introduces a new feature instead of fixing a bug or fixing a security vulnerability.

**Comment by tmplt** [comment]
Alright, understood. Is it possible to locally override the expression for a singular module so that the remainer of the system stays on a release channel?

**Comment by dasJ** [comment]
Yes! It's probably documented somewhere but I don't know where. You can just place this module anywhere and import it like `imports = [ ./modules/smartd.nix ];`. This would not work because all options defined in the nixos module will conflict with your options here. You can fix that by disabling the module from nixos using `disabledModules = [ "services/monitoring/smartd.nix" ];`

**Review by jonringer — APPROVED** [review]
diff LGTM

I agree with @dasJ that the backport shouldn't happen, but looks good for master

**Comment by rtldg** [comment]
The option description breaks building the docbook. 
```
...
building '/nix/store/vzp0b80cbdbpnq5cxaj6z6kh543d1s12-options-docbook.xml.drv'...
intermediate.xml:51743: parser error : Opening and ending tag mismatch: literal line 51743 and email
Acts as the value of <literal>email</email> in the emails' <literal>From: ... <e
                                           ^
intermediate.xml:51743: parser error : Opening and ending tag mismatch: email line 51743 and literal
ue of <literal>email</email> in the emails' <literal>From: ... <email></literal>
...
```

It should probably have `Acts as the value of <literal>email</literal> in the emails' <literal>From: ... &lt;email&gt;</literal> field.`

**Comment by jonringer** [comment]
@rtldg how do I build docbook?

**Comment by jonringer** [comment]
figured it out `nix-build nixos/release.nix -A manual.x86_64-linux`

**Pull request merged**